internal/ingress/controller/endpointslices.go

```diff
@@ -111,12 +112,38 @@ func getEndpointsFromSlices(s *corev1.Service, port *corev1.ServicePort, proto c
 				ports = append(ports, targetPort)
 			}
 		}
+		useTopologyHints = false
+		if zoneForHints != emptyZone {
+			useTopologyHints = true
```

Comment:
Can we also feature flag this? I guess changing the behavior of ingress directly to respect topology hints is not quite a breaking change, but at least a behavior change. 

I would like to allow someone to globally disable this behavior if something goes wrong (like ConfigMap or a flag)

Reply:
Sure, why not. Basic safeguard is not to have, annotated service, but i will add  configuration flag.

Reply:
done